pkr/ext/__init__.py

```diff
@@ -54,41 +57,52 @@ class Extensions(object):
     extensions
     """
 
-    def __init__(self, features):
+    def __init__(self, features=None):
         super(Extensions, self).__init__()
-        self.features = set(features)
+        self.features = features
         self._extensions = None
 
     @property
     def extensions(self):
         """Lazy loading of extensions"""
         if self._extensions is None:
-            self._extensions = NamedExtensionManager(
-                namespace='extensions',
-                names=self.features,
-                name_order=True,
-                propagate_map_exceptions=True
-            )
+            self._extensions = self.list_all()
+        if self.features is not None:
+            return {name: self._extensions[name] for name in self.features if name in self._extensions}
         return self._extensions
 
     def __getattr__(self, attribute):
-        if hasattr(ExtMixin, attribute):
-            if not self.features:
-                return lambda *args, **kw: ()
-            return lambda *args, **kw: self.extensions.map(
-                self._wrap_call, attribute, *args, **kw)
-        return super(Extensions, self).__getattribute__(attribute)
+        """Return all matching extensions methods wrapped into a lambda"""
+        # Unknown extension method
+        if not hasattr(ExtMixin, attribute):
+            return super(Extensions, self).__getattribute__(attribute)
+
+        # Called outside a kard, doing nothing
+        if not self.features:
+            return lambda *args, **kw: ()
+
+        def wrapper(*args, **kwargs):
+            return [
+                output[1]
+                for output in map(
+                    lambda ext: self._wrap_call(ext[0], ext[1], attribute, *args, *kwargs),
+                    self.extensions.items()
+                )
+                if output is not None
+            ]
+        return wrapper
 
     @staticmethod
-    def _wrap_call(extension, method_name, *args, **kwargs):
-        method = getattr(extension.plugin(), method_name, None)
+    def _wrap_call(name, extension, method_name, *args, **kwargs):
+        method = getattr(extension, method_name, None)
+        base_method = getattr(ExtMixin, method_name, None)
 
-        if method is None:
+        if method is None or method == base_method:
```

Comment:
here you can use `is` insteadof `==` 
We don't know if method and base_method have the same value but if method is base _method.

Reply:
Updated.